File: list_chats.py
"""Print every chat this Telegram account can see, with the ID to use
in TARGET_CHATS. Run: .venv/bin/python list_chats.py
"""
from telethon.sync import TelegramClient

import config

with TelegramClient(
    config.TELEGRAM_SESSION, config.TELEGRAM_API_ID, config.TELEGRAM_API_HASH
) as client:
    print(f"{'ID':>15}  NAME")
    print("-" * 60)
    for dialog in client.iter_dialogs():
        kind = "channel" if dialog.is_channel else "group" if dialog.is_group else "dm"
        print(f"{dialog.id:>15}  [{kind}] {dialog.name}")

File: src/listChats.ts
/** Print every chat this Telegram account can see, with the ID to use
 * in TARGET_CHATS. Run: npm run chats
 */
import readline from "node:readline/promises";

import { TelegramClient } from "telegram";
import { StoreSession } from "telegram/sessions/index.js";

import { config } from "./config.js";

async function main(): Promise<void> {
  const client = new TelegramClient(
    new StoreSession(config.TELEGRAM_SESSION),
    config.TELEGRAM_API_ID,
    config.TELEGRAM_API_HASH,
    { connectionRetries: 5 }
  );

  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
  await client.start({
    phoneNumber: () => rl.question("Phone number (international format): "),
    password: () => rl.question("2FA password: "),
    phoneCode: () => rl.question("Login code you received: "),
    onError: async (err) => {
      console.error(err);
      return true;
    },
  });
  rl.close();

  console.log(`${"ID".padStart(15)}  NAME`);
  console.log("-".repeat(60));
  for (const dialog of await client.getDialogs()) {
    const kind = dialog.isChannel ? "channel" : dialog.isGroup ? "group" : "dm";
    console.log(`${String(dialog.id).padStart(15)}  [${kind}] ${dialog.title}`);
  }
  await client.disconnect();
  process.exit(0);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});
